lib/PrintAsObjC/DeclAndTypePrinter.cpp

```diff
@@ -1137,22 +1137,19 @@ class DeclAndTypePrinter::Implementation
       }
 
       auto nominal = copyTy->getNominalOrBoundGenericNominal();
-      if (nominal && isa<StructDecl>(nominal)) {
-        if (nominal == ctx.getArrayDecl() ||
-            nominal == ctx.getDictionaryDecl() ||
-            nominal == ctx.getSetDecl() ||
-            nominal == ctx.getStringDecl() ||
-            (!getKnownTypeInfo(nominal) && getObjCBridgedClass(nominal))) {
-          // We fast-path the most common cases in the condition above.
-          os << ", copy";
-        } else if (nominal == ctx.getUnmanagedDecl()) {
-          os << ", unsafe_unretained";
-          // Don't print unsafe_unretained twice.
-          if (auto boundTy = copyTy->getAs<BoundGenericType>()) {
-            ty = boundTy->getGenericArgs().front();
-            if (isOptional)
-              ty = OptionalType::get(ty);
-          }
+      if (nominal && isa<StructDecl>(nominal) &&
+          (copyTy->isArray() || copyTy->isDictionary() || copyTy->isSet() ||
+          copyTy->isString() ||
+          (!getKnownTypeInfo(nominal) && getObjCBridgedClass(nominal)))) {
```

Comment:
Please put this back as a nested `if` to not mix `&&` and `||` even further. (The paren alignment got messed up and that made me double-take as it was.)

Reply:
oops I forgot this one, thanks!